src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +165,999 @@ static DamageTable D_80A63CE0 = {
+        } else {
+            this->timer++;
+        }
+    }
+}
+
+void EnBal_SetupFallFromSky(EnBal* this) {
+    Actor_PlaySfx(&this->picto.actor, NA_SE_VO_TIVO00);
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_FALL_LOOP);
+    this->forceEyesShut = true;
+    this->lastBalloonAction = EN_BAL_BALACT_FALL;
+    this->actionFunc = EnBal_FallFromSky;
+}
+
+void EnBal_FallFromSky(EnBal* this, PlayState* play) {
+    Vec3f worldPos;
+
+    worldPos = this->picto.actor.world.pos;
+    this->picto.actor.focus.pos.x = this->picto.actor.world.pos.x;
+    this->picto.actor.focus.pos.y = this->picto.actor.world.pos.y + 20.0f;
+    this->picto.actor.focus.pos.z = this->picto.actor.world.pos.z + 30.0f;
+    if (this->timer == 30) {
+        this->picto.actor.gravity = -1.5f;
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 5, 0x3000, 0x100);
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.z, this->picto.actor.world.rot.z, 5, 0x3000, 0x100);
+    } else if (this->timer < 30) {
+        this->timer++;
+        this->picto.actor.shape.rot.y = (s32)(Math_CosS(this->skyFloatPhase) * 2500.0f) + this->picto.actor.world.rot.y;
+        this->picto.actor.shape.rot.z = (s32)(Math_SinS(this->skyFloatPhase) * 2500.0f) + this->picto.actor.world.rot.z;
+        if (this->timer == 25) {
+            this->forceEyesShut = false;
+            this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_FALL_ONCE);
+        } else if (this->timer == 29) {
+            this->picto.actor.velocity.y = -12.0f;
+        }
+    }
+
+    if (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND_TOUCH) {
+        Actor_PlaySfx(&this->picto.actor, NA_SE_VO_TIVO01);
+        Actor_PlaySfx(&this->picto.actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
+
+    if (this->picto.actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
+        this->picto.actor.colChkInfo.health = 0;
+        if (this->timer >= 81) {
+            this->picto.actor.shape.yOffset = 0.0f;
+            EnBal_SetupGroundIdle(this);
+        } else if (this->timer == 30) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_LAND);
+            this->picto.actor.shape.rot = this->picto.actor.world.rot;
+            Actor_SpawnFloorDustRing(play, &this->picto.actor, &worldPos, 10.0f, 30, 5.0f, 0, 0, 0);
+            this->timer++;
+        } else {
+            if (play->gameplayFrames & 1) {
+                this->picto.actor.shape.yOffset = 20.0f;
+            } else {
+                this->picto.actor.shape.yOffset = 0.0f;
+            }
+            this->timer++;
+        }
+    } else if (this->picto.actor.velocity.y < 0.0f) {
+        Actor_PlaySfx_Flagged(&this->picto.actor, 0x20A0);
+    }
+
+    this->skyFloatPhase += 3000;
+}
+
+void EnBal_SetupInflateBalloon(EnBal* this) {
+    Vec3f scale;
+
+    scale = sInitBalloonScale;
```

Comment:
```suggestion
    Vec3f scale = {0.0f, 0.0f, 0.0f}

```
This looks like it was probably declared on the stack. (you would also need to delete `sInitBalloonScale` above)